ethereum/contracts/relayer/deliveryProvider/DeliveryProvider.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.19;
+
+import "./DeliveryProviderGovernance.sol";
+import "./DeliveryProviderStructs.sol";
+import "../../interfaces/relayer/IDeliveryProviderTyped.sol";
+import "../../interfaces/relayer/TypedUnits.sol";
+import "../../libraries/relayer/ExecutionParameters.sol";
+import {IWormhole} from "../../interfaces/IWormhole.sol";
+
+contract DeliveryProvider is DeliveryProviderGovernance, IDeliveryProvider {
+    using WeiLib for Wei;
+    using GasLib for Gas;
+    using GasPriceLib for GasPrice;
+    using WeiPriceLib for WeiPrice;
+    using TargetNativeLib for TargetNative;
+    using LocalNativeLib for LocalNative;
+
+    error CallerNotApproved(address msgSender);
+
+    function quoteEvmDeliveryPrice(
+        uint16 targetChain,
+        Gas gasLimit,
+        TargetNative receiverValue
+    )
+        public
+        view
+        returns (LocalNative nativePriceQuote, GasPrice targetChainRefundPerUnitGasUnused)
+    {
+        targetChainRefundPerUnitGasUnused = gasPrice(targetChain);
+        Wei costOfProvidingFullGasLimit = gasLimit.toWei(targetChainRefundPerUnitGasUnused);
+        Wei transactionFee =
+            quoteDeliveryOverhead(targetChain) + gasLimit.toWei(quoteGasPrice(targetChain));
+        Wei receiverValueCost = quoteAssetCost(targetChain, receiverValue);
+        nativePriceQuote = (
+            transactionFee.max(costOfProvidingFullGasLimit) + receiverValueCost
```

Comment:
We're taking the max(transactionFee, costOfProvidingFullGasLimit) but it appears they are in terms of two different native currencies.